components/script/dom/htmlimageelement.rs

```diff
@@ -1032,3 +1052,216 @@ fn image_dimension_setter(element: &Element, attr: LocalName, value: u32) {
+                        Some((idx, c)) => {
+                            state = ParseState::InDescriptor;
+                            buffered = Some((idx, c));
+                            continue;
+                        }
+                        None => {
+                            if !current_descriptor.is_empty() {
+                                descriptors.push_back(current_descriptor.clone());
+                            }
+                            break;
+                        }
+                    }
+                }
+            }
+        }
+        let mut error = false;
+        let mut width: Option<u32> = None;
+        let mut density: Option<f64> = None;
+        let mut future_compat_h: Option<u32> = None;
+        for descriptor in descriptors {
+            let char_iter = descriptor.chars();
+            let (digits, remaining) = collect_sequence_characters(&descriptor, is_ascii_digit);
+            let valid_non_negative_integer = parse_unsigned_integer(digits.chars());
+            let has_w = if remaining == "w" {
+                true
+            } else {
+                false
+            };
+            let valid_floating_point = parse_double(digits);
+            let has_x = if remaining == "x" {
+                true
+            } else {
+                false
+            };
+            let has_h = if remaining == "h" {
+                true
+            } else {
+                false
+            };
+            if valid_non_negative_integer.is_ok() && has_w {
+                //not support sizes attribute
+                if width.is_some() && density.is_some() {
+                    error = true;
+                }
+                let result = parse_unsigned_integer(char_iter.clone());
+                error = result.is_err();
+                if let Ok(w) = result {
+                    width = Some(w);
+                }
+            } else if valid_floating_point.is_ok() && has_x {
+                if width.is_some() && density.is_some() && future_compat_h.is_some() {
+                    error = true;
+                }
+                let result = parse_double(char_iter.as_str());
+                error = result.is_err();
+                if let Ok(x) = result {
+                    density = Some(x);
+                }
+            } else if valid_non_negative_integer.is_ok() && has_h {
+                if density.is_some() && future_compat_h.is_some() {
+                    error = true;
+                }
+                let result = parse_unsigned_integer(char_iter.clone());
+                error = result.is_err();
+                if let Ok(h) = result {
+                    future_compat_h = Some(h);
+                }
+            } else {
+                error = true;
+            }
+        }
+        if future_compat_h.is_some() && width.is_none() {
+            error = true;
+        }
+        if error == false {
+            if width.is_some() || density.is_some() {
```

Comment:
According to Step 15 of [spec](https://html.spec.whatwg.org/multipage/images.html#parsing-a-srcset-attribute): is written in a misleading way. Due to this check, the case of a single srcset url without any descriptor/density : (`small_image.jpg`), does not parses correctly and returns an empty vec. This is a valid srcset string as confirmed by neha.

The spec does not mean that we should append Image Source only if width and density is present.
Rather it should just check if error is false and just append whatever is there in the `width` and `density` variable along with url.

So basically this : 
```rust
if !error {
// append stuff to candidates
} else {
println!("parsed error");
}
```